cylc/flow/network/server.py

```diff
@@ -148,89 +149,122 @@ def __init__(self, schd, context=None, barrier=None,
             IgnoreFieldMiddleware,
         ]
 
-    def _socket_options(self):
-        """Set socket options.
+        self.queue = Queue()
+        self.publish_queue = Queue()
+        self.stopping = False
+        self.stopped = True
 
-        Overwrites Base method.
+    def configure(self):
+        self.register_endpoints()
 
-        """
-        # create socket
-        self.socket.RCVTIMEO = int(self.RECV_TIMEOUT) * 1000
+        # TODO: this in zmq asyncio context?
+        # Requires the scheduler main loop in asyncio first
+        # And use of concurrent.futures.ThreadPoolExecutor?
+        self.zmq_context = zmq.Context()
+        # create an authenticator for the ZMQ context
+        self.curve_auth = ThreadAuthenticator(self.zmq_context, log=LOG)
+        self.curve_auth.start()  # start the authentication thread
+
+        # Setting the location means that the CurveZMQ auth will only
+        # accept public client certificates from the given directory, as
+        # generated by a user when they initiate a ZMQ socket ready to
+        # connect to a server.
+        workflow_srv_dir = workflow_files.get_workflow_srv_dir(
+            self.schd.workflow)
+        client_pub_keyinfo = workflow_files.KeyInfo(
+            workflow_files.KeyType.PUBLIC,
+            workflow_files.KeyOwner.CLIENT,
+            workflow_srv_dir=workflow_srv_dir)
+        self.client_pub_key_dir = client_pub_keyinfo.key_path
+
+        # Initial load for the localhost key.
+        self.curve_auth.configure_curve(
+            domain='*',
+            location=(self.client_pub_key_dir)
+        )
 
-    def _bespoke_start(self):
-        """Setup start items, and run listener.
+        self.replier = WorkflowReplier(self, context=self.zmq_context)
+        self.publisher = WorkflowPublisher(self, context=self.zmq_context)
 
-        Overwrites Base method.
+    def start(self, barrier):
+        """Start the TCP servers."""
+        # set asyncio loop on thread
+        try:
+            self.loop = asyncio.get_running_loop()
+        except RuntimeError:
+            self.loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(self.loop)
+
+        min_, max_ = glbl_cfg().get(['scheduler', 'run hosts', 'ports'])
+        self.replier.start(min_, max_)
+        self.publisher.start(min_, max_)
+        self.port = self.replier.port
+        self.pub_port = self.publisher.port
+        self.schd.data_store_mgr.delta_workflow_ports()
+
+        # wait for threads to setup socket ports before continuing
+        barrier.wait()
+
+        self.stopped = False
+
+        self.operate()
+
+    async def stop(self, reason):
+        """Stop the TCP servers, and clean up authentication."""
+        self.queue.put('STOP')
+        if self.thread and self.thread.is_alive():
+            while not self.stopping:
+                sleep(self.STOP_SLEEP_INTERVAL)
```

Comment:
I think this should be an async sleep to avoid blocking, unless there is a specific reason to block?
```suggestion
                await asyncio.sleep(self.STOP_SLEEP_INTERVAL)
```

Reply:
`time.sleep` is used to yield control to other threads (halting the current thread).. Whereas, I believe `asyncio.sleep` just allows other ayncio execution for the duration.

Reply:
OK, would be good to have a comment mentioning this somewhere

Reply:
I put more into the docstring of the method.